cypress/e2e/po/pages/extensions.po.ts

```diff
@@ -0,0 +1,249 @@
+    this.enterGitRepoName('https://github.com/rancher/ui-plugin-examples');
+    this.enterGitBranchName('main');
+
+    // save it
+    return this.saveNewClusterRepoClick();
+  }
+
+  // extension card
+  extensionCard(extensionName: string) {
+    return this.self().getId(`extension-card-${ extensionName }`);
+  }
+
+  extensionCardClick(extensionName: string): Cypress.Chainable {
+    return this.extensionCard(extensionName).click();
+  }
+
+  extensionCardInstallClick(extensionName: string): Cypress.Chainable {
+    return this.extensionCard(extensionName).getId(`extension-card-install-btn-${ extensionName }`).click();
+  }
+
+  extensionCardUpdateClick(extensionName: string): Cypress.Chainable {
+    return this.extensionCard(extensionName).getId(`extension-card-update-btn-${ extensionName }`).click();
+  }
+
+  extensionCardRollbackClick(extensionName: string): Cypress.Chainable {
+    return this.extensionCard(extensionName).getId(`extension-card-rollback-btn-${ extensionName }`).click();
+  }
+
+  extensionCardUninstallClick(extensionName: string): Cypress.Chainable {
+    return this.extensionCard(extensionName).getId(`extension-card-uninstall-btn-${ extensionName }`).click();
+  }
+
+  // extension install modal
+  extensionInstallModal() {
+    return this.self().get('[data-modal="installPluginDialog"]');
+  }
+
+  installModalSelectVersionClick(optionIndex: number): Cypress.Chainable {
+    this.extensionInstallModal().getId('install-ext-modal-select-version').click();
+
+    return this.self().get(`.vs__dropdown-menu .vs__dropdown-option:nth-child(${ optionIndex })`).click();
+  }
+
+  installModalCancelClick(): Cypress.Chainable {
+    return this.extensionInstallModal().getId('install-ext-modal-cancel-btn').click();
+  }
+
+  installModalInstallClick(): Cypress.Chainable {
+    return this.extensionInstallModal().getId('install-ext-modal-install-btn').click();
+  }
+
+  // extension uninstall modal
+  extensionUninstallModal() {
+    return this.self().get('[data-modal="uninstallPluginDialog"]');
+  }
+
+  uninstallModalCancelClick(): Cypress.Chainable {
+    return this.extensionUninstallModal().getId('uninstall-ext-modal-cancel-btn').click();
+  }
+
+  uninstallModaluninstallClick(): Cypress.Chainable {
+    return this.extensionUninstallModal().getId('uninstall-ext-modal-uninstall-btn').click();
+  }
+
+  // extension details
+  extensionDetails() {
+    return this.self().getId('extension-details');
+  }
+
+  extensionDetailsBgClick(): Cypress.Chainable {
+    return this.self().getId('extension-details-bg').click();
+  }
+
+  extensionDetailsTitle(): Cypress.Chainable<string> {
+    return this.extensionDetails().getId('extension-details-title').invoke('text');
+  }
+
+  extensionDetailsCloseClick(): Cypress.Chainable {
+    return this.extensionDetails().getId('extension-details-close').click();
+  }
+
+  // extension tabs
+  extensionTabs() {
+    return this.self().getId('extension-tabs');
```

Comment:
We should PO the `Tabbed` and `Tab` components and use here

Reply:
At the moment your `TabPo` is more of a `TabbedPo` (`extension-tabs` links to a `Tabbed` component and has a nice function to select one of the child tabs). I'd vote the filename just gets renamed, `extensionTabs` then returns the `Tabbed` element and `extensionTabInstalledClick` calls clickNthTab from the TabbedPo component